File: 005/005.rb
require_relative "in_triangle"

print "A(xA, yA)? "
xA, yA = gets.split.map(&:to_f)
print "B(xB, yB)? "
xB, yB = gets.split.map(&:to_f)
print "C(xC, yC)? "
xC, yC = gets.split.map(&:to_f)
print "M(xM, yM)? "
xM, yM = gets.split.map(&:to_f)

sf = S_triangle(xA, yA, xB, yB, xC, yC)
s1 = S_triangle(xM, yM, xB, yB, xC, yC)
s2 = S_triangle(xM, yM, xA, yA, xC, yC)
s3 = S_triangle(xM, yM, xB, yB, xA, yA)

check_in(s1, s2, s3, sf)

File: 005/005.py
import in_triangle

xA, yA = map(float, input("A(xA, yA)? ").split())
xB, yB = map(float, input("B(xB, yB)? ").split())
xC, yC = map(float, input("C(xC, yC)? ").split())
xM, yM = map(float, input("M(xM, yM)? ").split())

Sf = in_triangle.S_triangle(xA, yA, xB, yB, xC, yC)
S1 = in_triangle.S_triangle(xM, yM, xB, yB, xC, yC)
S2 = in_triangle.S_triangle(xM, yM, xA, yA, xB, yB)
S3 = in_triangle.S_triangle(xM, yM, xA, yA, xC, yC)

in_triangle.check_in(S1, S2, S3, Sf)
